npc_combat_test.py

```diff
@@ -77,6 +77,23 @@ def test_clash_npc_mod(npc_mod, player_atk, npc_blk, expected_player_damage):
     npc = Character()
     player_hp_before_clash = player.hp
 
-    clash(player_atk, "LO", player, "LO", npc_blk, npc_mod, npc)
+    clash(player_atk, "LO", "foo", player, "LO", npc_blk, npc_mod, npc)
 
     assert player_hp_before_clash - player.hp == expected_player_damage
```

Comment:
I forgot to go back and implement the change you made me thought of: change this assertion to

`assert player.hp == player_hp_before_clash - expected_player_damage`

Again this more clearly separates the "actual" from the expected. For instructional purposes, this could be more verbosely expressed as:
```
actual = player.hp
expected = player_hp_before_clash - expected_player_damage
assert actual == expected
```

("What is the actual thing we're checking?" "The player.hp" "What is the expected value that we should see?" "We expect that their current hit points -- the "actual" -- should be equivalent to their hit points before the clash minus the expected amount of damage they should take from a clash which varies from test to test depending on the parameterized test values.")